src/CoreArbiterServer.cc

```diff
@@ -805,7 +805,10 @@ CoreArbiterServer::timeoutThreadPreemption(int timerFd)
 void
 CoreArbiterServer::cleanupConnection(int socket)
 {
-    sys->close(socket);
+    if (sys->close(socket) == -1) {
+        // The socket was closed before, just return
```

Comment:
Aren't there many different errors that close can return? How do you know that it failed because the socket was already closed, as opposed to some other cause? Also, shouldn't you know whether you have already closed the socket?  Let's discuss.